tlatools/org.lamport.tlatools/src/tla2sany/utilities/Vector.java

```diff
@@ -1,143 +1,100 @@
-    return size;
+  public Vector(int initialSize) {
+    super(initialSize);
   }
 
-  /* Method replaced below
-  public final String toString() {
-    String ret="";
-
-    for (int i = 0; i < size; i++) {
-      ret = ret + info[i].toString();
-    }
-
-    return ret;
-  }
-  */
-
-  public final void addElement( E obj ) {
-    if (size == capacity) {
-      Object next[] = new Object[ capacity + increment ];
-      System.arraycopy( info, 0, next, 0, capacity );
-      capacity+= increment;
-      info = next;
-    }
-    info[ size ] = obj;
-    size++;
+  public final void addElement(E obj) {
+    super.add(obj);
   }
 
-  @SuppressWarnings("unchecked")
   public final E firstElement() {
-    return (E)info[0];
+    return elementAt(0);
   }
 
-  @SuppressWarnings("unchecked")
   public final E lastElement() {
-    return (E)info[ size-1 ];
+    return elementAt(size() - 1);
   }
 
-  @SuppressWarnings("unchecked")
   public final E elementAt(int i) {
-    if (i < 0 || i >= size )
-      throw new ArrayIndexOutOfBoundsException();
-    else
-      return (E)info[ i ];
+    checkBounds(i);
+    return super.get(i);
   }
 
   public final void removeAllElements() {
-    for (int lvi = 0; lvi < size; lvi++ ) {
-      info[lvi] = null;
-    }
-    size = 0;
+    super.clear();
   }
 
-  public final void removeElementAt( int i ) {
-    if (i < 0 || i >= size )
-      throw new ArrayIndexOutOfBoundsException();
-    else {
-      for (int lvi = i+1; lvi < size; lvi++ )
-        info[ lvi-1 ] = info [lvi];
-      size--;
-      info[ size ] = null;
-    }
+  public final void removeElementAt(int i) {
+    checkBounds(i);
+    super.remove(i);
   }
 
-  public final void insertElementAt( E obj, int i ) {
-    if (i < 0 || i >= size )
+  public final void insertElementAt(E obj, int i) {
+    // The legacy implementation forbids inserting at the end (i == size()).
+    if (i < 0 || i >= size()) {
       throw new ArrayIndexOutOfBoundsException();
-    else if (size == capacity) {
-      Object next[] = new Object[ capacity + increment ];
-      System.arraycopy( info, 0, next, 0, i );
-      next[i] = obj;
-      System.arraycopy( info, i, next, i+1, capacity - i );
-      capacity+= increment;
-      info = next;
-    } else {
-      for ( int lvi = size; lvi > i; lvi-- )
-        info[ lvi ] = info[lvi-1];
-      info[ i ] = obj;
     }
-    size++;
+    super.add(i, obj);
   }
 
-  public final void setElementAt( E obj, int i ) {
-    if (i < 0 || i >= size )
-      throw new ArrayIndexOutOfBoundsException();
-    else 
-      info[ i ] = obj;
+  public final void setElementAt(E obj, int i) {
+    checkBounds(i);
+    super.set(i, obj);
   }
 
-  public final boolean contains (E obj) {
-    for (int i = 0; i < size; i++) {
-      if ( info[ i ] == obj ) return true;
+  @Override
+  public final boolean contains(Object obj) {
+    for (Iterator<E> iterator = super.iterator(); iterator.hasNext();) {
+      if (iterator.next() == obj) {
```

Comment:
This can be a bit simpler:
```java
for (E value : this) {
  if (value == obj) {
    ...
```
For-each loops desugar to almost exactly the code you have here.